corehq/apps/app_manager/static/app_manager/js/app_manager_media.js

```diff
@@ -1,150 +1,190 @@
+                replace_attachment: true
+            };
         };
-    };
 
-    self.getControllerRef = function () {
-        return {
-            path: self.currentPath(),
-            isMediaMatched: self.isMediaMatched,
-            getUrl: self.url,
-            m_id: self.objectId
+        self.getControllerRef = function () {
+            return {
+                path: self.currentPath(),
+                isMediaMatched: self.isMediaMatched,
+                getUrl: self.url,
+                m_id: self.objectId
+            };
+        };
+
+        self.passToUploadController = function () {
+            self.uploadController.resetUploader();
+            self.uploadController.currentReference = self.getControllerRef();
+            self.uploadController.uploadParams = self.getUploadParams();
+            self.uploadController.updateUploadFormUI();
+        };
+
+        self.uploadComplete = function (trigger, event, data) {
+            if (data.ref) {
+                var ref = data.ref;
+                var obj_map = self.objectMap();
+                obj_map[ref.path] = ref;
+                self.ref(new MenuMediaReference(ref));
+                self.objectMap(obj_map);
+                self.updateResource();
+                if (self.currentPath() !== data.ref.path){
+                    //CurrentPath has a different filetype to the
+                    //uploaded file
+                    self.customPath(data.ref.path);
+                }
+            }
+        };
+
+        self.updateResource = function () {
+            self.inputElement.trigger('change');
         };
     };
 
-    self.passToUploadController = function () {
-        self.uploadController.resetUploader();
-        self.uploadController.currentReference = self.getControllerRef();
-        self.uploadController.uploadParams = self.getUploadParams();
-        self.uploadController.updateUploadFormUI();
+    var MenuMediaReference = function (ref) {
+        'use strict';
+        var self = this;
+
+        self.path = ref.path || '';
+        self.iconType = ref.icon_class || '';
+        self.mediaType = ref.media_class || '';
+        self.module = ref.module;
+        self.form = ref.form;
     };
 
-    self.uploadComplete = function (trigger, event, data) {
-        if (data.ref) {
-            var ref = data.ref;
-            var obj_map = self.objectMap();
-            obj_map[ref.path] = ref;
-            self.ref(new MenuMediaReference(ref));
-            self.objectMap(obj_map);
-            self.updateResource();
-            if (self.currentPath() !== data.ref.path){
-                //CurrentPath has a different filetype to the
-                //uploaded file
-                self.customPath(data.ref.path);
-            }
+    function initNavMenuMedia(qualifier, imageRef, audioRef, objectMap, defaultFileName) {
+        var uploaders = hqImport('#app_manager/partials/nav_menu_media_js_common.html');
```

Comment:
Ignore me if I am too late for this party. My vote would be for `file/name.html#modulename` since that that allows us to intuitively define multiple modules per HTML file (each script block could potentially have its own module definition).